superset/utils/core.py

```diff
@@ -985,11 +988,11 @@ def is_adhoc_metric(metric) -> bool:
     )
 
 
-def get_metric_name(metric):
-    return metric["label"] if is_adhoc_metric(metric) else metric
+def get_metric_name(metric: Metric) -> str:
```

Comment:
@villebro the code works if the `metric` is `None` even though the `Metric` type is explicitly `Union[Dict[str, str], str]`. Generally I think we should strive to ensure only valid inputs are provided to functions (otherwise the logic can get quite complex), i.e., I prefer, 

```python
if metric:
    name = get_metric_name(metric)
    ...
```

rather than, 

```python
name = get_metric_name(metric)

if name:
   ...
```

Reply:
I agree.